cmd/server/main.go

```diff
@@ -60,6 +63,11 @@ func init() {
 	flag.StringVar(&AuthzAddr, "authz", "", "address of the authorization service")
 	flag.StringVar(&LogLevel, "log", "info", "log level")
 	flag.Parse()
+
+	resource.RegisterCodec(integer.NewCodec("contact-app", "contacts"), &pb.Contact{})
```

Comment:
so, explain to me again how this is supposed to work... every app that has a resource needs to register a `codec` for every type it wants to deal with? This seems kind of crazy if true...

Reply:
It is true.

We need to have an unified representation in PB.

For each proto message we need a way  to define correct resource name, so we cannot rely on the name of proto message itself.

For each “reference” we need to attach application name.

The “reference” must be able to be stored in DB in any format defined by an application.

So if you know how to implement all these requirement in less crazy way, please let me know.